Tools/CISupport/ews-build/steps.py

```diff
@@ -2150,6 +2174,188 @@ def hideStepIf(self, results, step):
+    name = 'filter-pull-requests-into-queues'
+
+    @defer.inlineCallbacks
+    def run(self):
+        num_prs = yield self.getNumPRs()
+        if num_prs:
+            sorted_prs = yield self.sortPRs(num_prs)
+            if sorted_prs:
+                list_of_prs, label_as_safe, label_as_blocked = sorted_prs
+            else:
+                return FAILURE
+            self.setProperty('safe_pr_list', label_as_safe)
+            yield self._addToLog('stdio', 'PRs to merge: {}.\n'.format(label_as_safe))
+            self.setProperty('blocked_pr_list', label_as_blocked)
+            yield self._addToLog('stdio', 'PRs to block: {}\n'.format(label_as_blocked))
+
+            if len(label_as_safe):
+                self.build.addStepsAfterCurrentStep([SafeMergeToMergeQueue()])
+            if len(label_as_blocked):
+                self.build.addStepsAfterCurrentStep([SafeMergeToMergeBlocked()])
+
+        defer.returnValue(SUCCESS if num_prs else FAILURE)
+
+    def getResultSummary(self):
+        if self.results == SUCCESS:
+            return {'step': f"Successfully requested and filtered pull requests"}
+        elif self.results == FAILURE:
+            return {'step': f"Failed to request and filter pull requests"}
+        return buildstep.BuildStep.getResultSummary(self)
+
+    @defer.inlineCallbacks
+    def getNumPRs(self, label='\"safe-merge-queue\"'):
+        query_body = '{ repository(owner:"Webkit", name:"WebKit") { pullRequests(labels: %s) { totalCount } } }' % label
+        query = {'query': query_body}
+        rc = yield self.query_graph_ql(query)
+        if rc:
+            num_prs = rc['data']['repository']['pullRequests']['totalCount']
+            yield self._addToLog('stdio', 'There are {} PR(s) in safe-merge-queue.\n'.format(num_prs))
+            defer.returnValue(num_prs)
+        else:
+            yield self._addToLog('stdio', 'Failed to retrieve number of PRs.\n')
+            defer.returnValue(None)
+
+    @defer.inlineCallbacks
+    def sortPRs(self, num_prs, label='\"safe-merge-queue\"'):
+        query_body = '{ repository(owner:"Webkit", name:"WebKit") { pullRequests(labels: %s, last: %s) { edges { node { title number commits(last: 3) { nodes { commit { commitUrl status { state contexts { context state } } } } } } } } } }' % (label, num_prs)
+        query = {'query': query_body}
+
+        yield self._addToLog('stdio', "Fetching all PRs with label {}...\n".format(label))
+
+        response = yield self.query_graph_ql(query)
+        if not response:
+            yield self._addToLog('stderr', 'Failed to retrieve list of PRs.\n')
+            defer.returnValue(None)
+
+        all_pr_data = response['data']['repository']['pullRequests']['edges']
+        list_of_prs = []
+        passed_status_check = []
+        failed_status_check = []
+        no_checks = []
+        ios_tvos_watchos = ['ios', 'mac', 'ios-sim', 'ios-wk2', 'ios-wk2-wpt', 'api-ios', 'tv', 'tv-sim', 'watch', 'watch-sim']
```

Comment:
We should move this list to more prominent place, atleast at the beginning of the class. Also make the variable name more descriptive. e.g.: QUEUES_FOR_SAFE_MERGE_QUEUE.

Also, I wonder if we should start with start a large list of queues, especially the testers queues like wk1, wk2, api tests) might have false positive. Can check if @JonWBedard @ryanhaddad have any strong opinion on it.